src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let row = '';
+    let sep = String(separator);
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1) {
+        sep = '';
+      };
+
+      if (typeof this[i] === 'undefined' || typeof this[i] === 'object') {
+        row += '' + sep;
+        continue;
+      };
+
+      row += this[i] + sep;
```

Comment:
```suggestion
    if (this[i] !== null && this[i] !== undefined) {
        row += this[i];
      }

     if (i !== this.length - 1) {
       row += `${separator}`;
     }
```

Reply:
the logic is that you add a current symbol and separator in different conditions one after another, do you understand now?

Reply:
Yes, got it finally.